packages/flutter_tools/lib/src/commands/multi_drive.dart

```diff
@@ -0,0 +1,375 @@
+
+  List<Device> _devices;
+
+  List<Device> get devices => _devices;
+
+  dynamic specs;
+
+  @override
+  Future<int> runInProject() async {
+    String specsPath = argResults['specs'];
+    if(specsPath == null) {
+      print('--specs=$specsPath, you must pass a non-null path to the specs argument.');
+      return 1;
+    }
+    this.specs = await _loadSpecs(specsPath);
+    print(specs);
+
+    this._devices = await targetDevicesFinder();
+    if (devices == null) {
+      return 1;
+    }
+
+    String testFile = specs['test-path'];
+
+    if (await fs.type(testFile) != FileSystemEntityType.FILE) {
+      printError('Test file not found: $testFile');
+      return 1;
+    }
+
+    if (!argResults['use-existing-app']) {
+      // printStatus('Starting application: ${argResults["target"]}');
+
+      if (getBuildMode() == BuildMode.release) {
+        // This is because we need VM service to be able to drive the app.
+        printError(
+          'Flutter Multi-Device Driver does not support running in release mode.\n'
+          '\n'
+          'Use --profile mode for testing application performance.\n'
+          'Use --debug (default) mode for testing correctness (with assertions).'
+        );
+        return 1;
+      }
+
+      int result = await appsStarter(this);
+      if (result != 0) {
+        printError('Application failed to start. Will not run test. Quitting.');
+        return result;
+      }
+    } else {
+      printStatus('Will connect to already running application instance.');
+    }
+
+    // Check for the existance of a `packages/` directory; pub test does not yet
+    // support running without symlinks.
+    if (!new io.Directory('packages').existsSync()) {
+      Status status = logger.startProgress(
+        'Missing packages directory; running `pub get` (to work around https://github.com/dart-lang/test/issues/327):'
+      );
+      await runAsync(<String>[sdkBinaryName('pub'), 'get', '--no-precompile']);
+      status.stop(showElapsedTime: true);
+    }
+
+    try {
+      return await testRunner(<String>[testFile])
+        .catchError((dynamic error, dynamic stackTrace) {
+          printError('CAUGHT EXCEPTION: $error\n$stackTrace');
+          return 1;
+        });
+    } finally {
+      if (!argResults['keep-app-running'] && !argResults['use-existing-app']) {
+        printStatus('Stopping application instance.');
+        try {
+          await allAppsStopper(this);
+        } catch(error, stackTrace) {
+          // TODO(yjbanov): remove this guard when this bug is fixed: https://github.com/dart-lang/sdk/issues/25862
+          printTrace('Could not stop application: $error\n$stackTrace');
+        }
+      } else {
+        printStatus('Leaving the application running.');
+      }
+    }
+  }
+
+  Future<dynamic> _loadSpecs(String specsPath) async {
+    // Read specs file into json format
+    dynamic spec = JSON.decode(await new io.File(specsPath).readAsString());
```

Comment:
1. `spec` -> `specs`, just to be consistent.
2. Some error handling should be done here. What if the file doesn't exist, for example?